### PR TITLE
[W5.7] [T9-1] Tan E-Liang

### PR DESCRIPTION
**Pull request by taneliang**
My name has a hyphen in it, but AddressBook doesn't support it. This PR rectifies the situation.

### CODE REVIEW

File: src/seedu/addressbook/data/person/Name.java [hunk 1]
```diff
@@ -22,7 +22,13 @@
      * @throws IllegalValueException if given name string is invalid.
      */
     public Name(String name) throws IllegalValueException {
-        String trimmedName = name.trim();
+        // Trim start/end spaces and hyphens
+        String trimmedName = "";
+        String trimmedAgainName = name;
+        while (!trimmedName.equals(trimmedAgainName)) {
+            trimmedName = trimmedAgainName;
+            trimmedAgainName = trimmedAgainName.trim().replaceAll("-$|^-", "");;
```
Comment (JayPeeTeeDee) [inline src/seedu/addressbook/data/person/Name.java]: Extra semicolon

File: test/java/seedu/addressbook/commands/AddCommandTest.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
 
     @Test
     public void addCommand_invalidName_throwsException() {
-        final String[] invalidNames = { "", " ", "[]\\[;]" };
+        final String[] invalidNames = { "", " ", "-", "- ", "- -- ---", "[]\\[;]" };
```
Comment (JayPeeTeeDee) [inline test/java/seedu/addressbook/commands/AddCommandTest.java]: Add additional edge case: " -"?

File: src/seedu/addressbook/data/person/Name.java (hunk 1 above)
Comment (taneliang) [reply src/seedu/addressbook/data/person/Name.java]: Oops

File: docs/UserGuide.adoc [hunk 3]
```diff
@@ -59,7 +59,8 @@ items with `...` after them can have multiple instances. Order of parameters are
 Put a `p` before the phone / email / address prefixes to mark it as `private`. `private` details can only
 be seen using the `viewall` command.
 
-Persons can have any number of tags (including 0).
+Persons can have any number of tags (including 0). Person names must only contain spaces, hyphens or letters,
+but cannot be just spaces or hyphens.
```
Comment (novellius) [inline docs/UserGuide.adoc:63]: Good that you've updated User Guide to reflect the change.

File: test/expected.txt [hunk 4]
```diff
@@ -74,7 +74,7 @@
 || Tags names should be alphanumeric
 || ===================================================
 || Enter command: || [Command entered:  add []\[;] p/12345 e/valid@e.mail a/valid, address]
-|| Person names should be spaces or alphabetic characters
+|| Person names should be spaces, hyphens or alphabetic characters
```
Comment (novellius) [inline test/expected.txt:77]: Good that you've updated the I/O tests, but perhaps you can write more test commands to test that names with hyphens are accepted by the AddressBook. You can also edit AddressBookTest to change some of the names to include hyphens.

File: src/seedu/addressbook/data/person/Name.java [hunk 5]
```diff
@@ -11,9 +11,9 @@
  */
 public class Name {
 
-    public static final String EXAMPLE = "John Doe";
-    public static final String MESSAGE_NAME_CONSTRAINTS = "Person names should be spaces or alphabetic characters";
-    public static final String NAME_VALIDATION_REGEX = "[\\p{Alpha} ]+";
+    public static final String EXAMPLE = "John-John Doe";
+    public static final String MESSAGE_NAME_CONSTRAINTS = "Person names should be spaces, hyphens or alphabetic characters";
```
Comment (novellius) [inline src/seedu/addressbook/data/person/Name.java:15]: Perhaps include the statement about how the names cannot be just hyphens and spaces? It would remind the user of the specific constraints.

**Review by novellius — APPROVED** [review]
Overall, good job @taneliang . See my comments for suggestions. Please close the PR after reading them.
